Add check for 404 on get repository data

dmcgowan opened the pull request:
No longer add the body to the error when a 404 is received on get repository data.

closes #11510

jfrazelle commented:
LGTM

ahmetalpbalkan commented on registry/session.go:
```diff
@@ -281,6 +281,9 @@ func (r *Session) GetRepositoryData(remote string) (*RepositoryData, error) {
 	// TODO: Right now we're ignoring checksums in the response body.
 	// In the future, we need to use them to check image validity.
 	if res.StatusCode != 200 {
+		if res.StatusCode == 404 {
```
I think this should be like 

```
if res.StatusCode == 404 { ...}
else if res.StatusCode != 200 { ... }
```

which reduces nesting = more readable. Also we should be using `http.StatusNotFound` `http.StatusOK` but this was before you I guess. :)

icecrime replied:
I agree with both remarks. @dmcgowan let us know if you can update, or we'll merge this as is for 1.6.0 RC1 and review later.

dmcgowan replied:
I will update the nesting. The use of constants over numbers should be done in a refactor PR, I will keep this change consistent with the rest of the code.

dmp42 commented:
LGTM

dmcgowan committed:
Add check for 404 on get repository data

No longer add the body to the error when a 404 is received on get repository data.

closes #11510

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

icecrime commented:
Code LGTM: waiting on CI

icecrime commented:
Green, thanks!
